app/src/main/java/com/geeksville/mesh/repository/usb/UsbRepository.kt

```diff
@@ -0,0 +1,77 @@
+package com.geeksville.mesh.repository.usb
+
+import android.app.Application
+import android.hardware.usb.UsbDevice
+import android.hardware.usb.UsbManager
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.coroutineScope
+import com.geeksville.android.Logging
+import com.geeksville.mesh.CoroutineDispatchers
+import com.hoho.android.usbserial.driver.UsbSerialProber
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Repository responsible for maintaining and updating the state of USB connectivity.
+ */
+@OptIn(ExperimentalCoroutinesApi::class)
+@Singleton
+class UsbRepository @Inject constructor(
+    private val application: Application,
+    private val dispatchers: CoroutineDispatchers,
+    private val processLifecycle: Lifecycle,
+    private val usbBroadcastReceiverLazy: dagger.Lazy<UsbBroadcastReceiver>,
+    private val usbManagerLazy: dagger.Lazy<UsbManager?>,
+    private val usbSerialProberLazy: dagger.Lazy<UsbSerialProber>
+) : Logging {
+    private val _serialDevices = MutableStateFlow(emptyMap<String, UsbDevice>())
+
+    @Suppress("unused") // Retained as public API
+    val serialDevices = _serialDevices
+        .asStateFlow()
+
+    val serialDevicesWithDrivers = _serialDevices
```

Comment:
Eventually it would be cool if the UI responded to this state flow reactively.  For now we just use it to hold the most recent state.